src/test/groovy/io/github/cosmicsilence/scalafix/ScalafixPluginFunctionalTest.groovy

```diff
@@ -13,22 +14,22 @@ class ScalafixPluginFunctionalTest extends Specification {
     public final TemporaryFolder testProjectDir = new TemporaryFolder()
     private File settingsFile
     private File buildFile
+    private File scalaSrcFile
+    private File scalafixConfFile
 
     def setup() {
         settingsFile = testProjectDir.newFile("settings.gradle")
         buildFile = testProjectDir.newFile("build.gradle")
 
-        buildFile.write'''
-plugins {
-    id 'scala'
-    id 'io.github.cosmicsilence.scalafix'
-}
-
-repositories {
-    mavenCentral()
-}
-'''
+        buildFile.write ScalafixTestConstants.BUILD_SCRIPT
         settingsFile.write "rootProject.name = 'hello-world'"
+
+        // write a minimal scala source file with an unused import
+        final File scalaSrcDir = testProjectDir.newFolder("src", "main", "scala", "io", "github", "cosmicsilence", "scalafix")
+        scalaSrcFile = new File(scalaSrcDir.absolutePath +  "/HelloWorld.scala")
+        scalaSrcFile.write ScalafixTestConstants.FLAWED_SCALA_CODE
+
+        scalafixConfFile = testProjectDir.newFile(ScalafixExtension.DEFAULT_CONFIG_FILE)
```

Comment:
it would be nice to have a utility method that took the content of the source file and created it at the appropriate location so we could play with different issues, multiple files and also with files that have no issue. Something along these lines:

```groovy
    private File createSourceFile(String content) {
        def scalaSrcDir = testProjectDir.newFolder("src", "main", "scala")
        def sourceFile = new File(scalaSrcDir.absolutePath,  "package.scala")
        sourceFile.write content
        sourceFile
    }
```

and then we can use in individual tests like so:

```groovy
    def 'scalafix should run semantic rule that fixes scala code'() {
        given:
        def inputSourceFile = createSourceFile('''
import scala.util.Random
object HelloWorld
''')
        scalafixConfFile.write "rules = [ RemoveUnused ]"

        when:
        runGradleTask('scalafix', [ ])

        then:
        inputSourceFile.getText() == '''
object HelloWorld
'''
    }
```

Reply:
cool i will inline the build script and add this `createSourceFile` method